### ISSUE TITLE
[Bug] PCC Drop at Runtime When SDPA Receives Slightly Larger Key Tensor in Qwen2.5-0.5B-Instruct Model

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-forge-fe#2557]
## Description

- Qwen2.5-0.5B-Instruct model facing `Tensor mismatch. PCC = 0.2453387212723644` 
- Issues arises from [scaled_dot_product_attention](https://github.com/huggingface/transformers/blob/51f94ea06d19a6308c61bbb4dc97c40aabd12bad/src/transformers/integrations/sdpa_attention.py#L54C19-L54C68) 
  - Case 1 - Able to replicate the drop (`0.2792242347052282`) while passing saved input( tensors saved during cpu forward run) to sdpa alone(sanity).
  - Case 2 - But same test case passed while passing random tensors( torch.randn())
  - On closer look I found that original key tensor has some big values (e.g., 66.6809, 116.0370), whereas torch.randn() generates values in a much smaller range.
  - Case 3 - So inputs are created in the following way to mimic original scenario and drop observed(`0.5085714282207656`) in this case also 

```
query = torch.randn(1, 14, 39, 64)
key = torch.randn(1, 14, 39, 64)* 50 + 10 # to mimic original scenario
value = torch.randn(1, 14, 39, 64)
causal_mask = torch.randn(1, 14, 39, 39)
```

- The TTNN graph of the sdpa has the following TTNN ops, where one or more may be failing to handle large tensor values correctly:

```
ttnn.get_device
ttnn.reshape
ttnn.zeros
ttnn.add
ttnn.deallocate
ttnn.matmul
ttnn.multiply
ttnn.softmax
```
These conclusions can be verified using the following steps.

## Reproduce

```
git checkout kkannan/jul14_qwen_2_5_pcc_drop
git submodule update --recursive
git lfs pull --include='causal_mask.pt key.pt query.pt value.pt'

## model run - 0.2453387212723644
pytest forge/test/models/pytorch/text/qwen/test_qwen_v2.py::test_qwen_clm[Qwen/Qwen2.5-0.5B-Instruct] -svv 

## sanity runs

# case 1 - 0.2792242347052282
pytest forge/test/models/pytorch/text/qwen/test_sanity_case_1.py -svv

# case 2 - PASS 
pytest forge/test/models/pytorch/text/qwen/test_sanity_case_2.py -svv

# case 3 - 0.5085714282207656
pytest forge/test/models/pytorch/text/qwen/test_sanity_case_3.py -svv

```

## Logs

- [jul14_qwen_pcc_drop.log](https://github.com/user-attachments/files/21214698/jul14_qwen_pcc_drop.log)
- [jul14_case_1.log](https://github.com/user-attachments/files/21214696/jul14_c1.log)
- [jul14_case_2.log](https://github.com/user-attachments/files/21214697/jul14_c2.log)
- [jul14_case_3.log](https://github.com/user-attachments/files/21214699/jul14_c3.log)

**Comment by kamalrajkannan78** [comment]
## Root cause 

from the results of case 2 & 3 [here](https://github.com/tenstorrent/tt-forge-fe/issues/2557#issue-3228356521), its clear that one of the mentioned ttnn ops fails to handle large key tensor. 
Further analysis confirms that the issue originates from the ttnn.softmax operator. - https://github.com/tenstorrent/tt-metal/issues/25193
softmax op fails to handle large input tensor and this issue can be cross verified with below forge sanity too.


## Sanity

```
@pytest.mark.push
@pytest.mark.parametrize(
    "input_tensor",
    [
        torch.randn(1, 14, 39, 39),  # PASSED
        torch.randn(1, 14, 39, 39) * 50 + 10,  # PCC = 0.24953964370191267
    ],
)
def test_softmax(input_tensor):
    class Softmax(nn.Module):
        def __init__(self):
            super().__init__()
            self.softmax = nn.Softmax(dim=-1)

        def forward(self, a):
            return self.softmax(a)

    inputs = [input_tensor]

    framework_model = Softmax()
    compiled_model = forge.compile(framework_model, sample_inputs=inputs)

    verify(inputs, framework_model, compiled_model)
```


## Reproduce

```
git checkout kkannan/jul16_qwen_softmax_drop
git submodule update --recursive
pytest forge/test/mlir/operators/nn/test_nn.py -k "softmax" -svv
```

## Logs

[jul16_softmax_issue.log](https://github.com/user-attachments/files/21263760/jul16_softmax_failure.log)
